app/src/main/java/ch/benediktkoeppel/code/droidplane/HorizontalMindmapView.java

```diff
@@ -604,7 +616,50 @@ private int getVisiblePixelOfLeftmostColumn() {
 
         return numVisiblePixelsOnColumn;
     }
+    
+    /** Show a dialog to input the search string and fires the search. */
+    void startSearch() {
+        AlertDialog.Builder alert = new AlertDialog.Builder(getContext());
+        EditText input = new EditText(getContext());
+        input.setText(lastSearchString, TextView.BufferType.EDITABLE);
+        alert.setView(input);
+        alert.setPositiveButton("OK", (dialog, which) -> search(input.getText().toString()));
+        alert.create().show();
+    }
+    
+    /** Performs the search, stores the result, and selects the first matching node. */
+    private void search(String searchString) {
+        lastSearchString = searchString;
+        var searchRoot = nodeColumns.get(nodeColumns.size() - 1).getParentNode();
+        searchResultNodes = searchRoot.search(searchString);
+        currentSearchResultIndex = 0;
+        showCurrentSearchResult();
+    }
+    
+    /** Selects the current search result node. */
+    private void showCurrentSearchResult() {
+        if (currentSearchResultIndex >= 0 && currentSearchResultIndex < searchResultNodes.size()) {
+            downTo(getContext(), searchResultNodes.get(currentSearchResultIndex));
```

Comment:
**For result nodes with no children:** I like that this highlights the found node. But when the result node has lots of siblings, it happens that the result is not visible / user needs to scroll down first to go find the node.  Can we vertically scroll the resulting node into view?

**For result nodes with children:** If a result node has children, it already navigates into the list of children. As a result, in vertical screen orientation (with only one node column visible), you don't actually see the resulting node, but its children. Can we instead highlight the resulting node, and then the user taps on it to navigate into it?

Reply:
Ah yes, I didn't test that properly, it needs to be fixed.